### DISCUSSION TITLE
Problem to injection javascript

### DISCUSSION THREAD

**Discussion by msarocha** [daijro/camoufox#137] [Q&A]
I am using javascript to locate the recaptcha callback function, however the "___grecaptcha_cfg.clients" object is not visible when I inject the code into the website, if I execute it directly by pasting from the console the object is returned. I thought it might be some site detection, but with seleniumbase the javascript injection works fine. Any idea what it could be?

**Comment by netdev1** [comment]
Probably related to main js world execution, read this: https://camoufox.com/python/main-world-eval/

**Comment by msarocha** [comment]
Your suggestion helped a lot because I think it has something to do with an antibot on this page, see the simple test I did:

with Camoufox(main_world_eval=True) as browser:
    page = browser.new_page()
    page.goto("https://patrickhlauke.github.io/recaptcha/")
    recaptcha_data = page.evaluate(f"mw:___grecaptcha_cfg.clients")
    print(f"{recaptcha_data=}") # ==> Ok on this site
    page.goto("https://canal360i.cloud.itau.com.br/login/iparceiros/")
    recaptcha_data = page.evaluate(f"mw:___grecaptcha_cfg.clients")
    print(f"{recaptcha_data=}") # ==> None on this site, but in console it's Okay, why?
